lib/spack/spack/schema/env.py

```diff
@@ -119,6 +119,13 @@
                                 }
                             }
                         ]
+                    },
+                    'concretization': {
+                        'type': 'string',
+                        # Strategies are added dynamically by the
```

Comment:
Are there any other instances of dynamic schema modification? I'd prefer to keep all schema details in the schema files themselves to avoid bouncing around files.

Reply:
Ok. The downside is that we have to repeat a few names twice - it surely is not a big deal here.

Reply:
Implemented in 2cffb2d. It requires more boilerplate to ensure the schema and `Environment` class are consistent.